### PR TITLE
Kconfig: add modules to test on native

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
This continues modelling modules in Kconfig and adds `native` to the same Kconfig tests that we run for the `samr21-xpro`.

### Testing procedure
- Green CI
- Check that the dependencies are correctly modelled
- Peripheral test applications should still work as usual when using `TEST_KCONFIG=1`
- Non-supported applications should explain the missing features/modules.
- Usage of Kconfig as usual (without `TEST_KCONFIG=1`) should still work normally

### Issues/PRs references
Related to #14055

### CODE REVIEW

File: drivers/periph_common/Kconfig [hunk 1]
```diff
@@ -125,6 +125,8 @@ config MODULE_PERIPH_INIT_RTC
     default y if MODULE_PERIPH_INIT
     depends on MODULE_PERIPH_RTC
 
+osource "$(RIOTCPU)/$(CPU)/periph/Kconfig.rtc"
```
Comment (fjmolinas) [inline drivers/periph_common/Kconfig:128]: can we be consistent here, the rest of the file uses `rsource "Kconfig.<periph>"`

**Review by fjmolinas — COMMENTED** [review]
Looks good first look, will take a closer look comparing both dependencies and play around a bit.

**Comment by fjmolinas** [comment]
Bumped into the following when compiling with `TEST_KCONFIG` 1, then 0, then 1:

```
TEST_KCONFIG=1 BOARD=native make -C tests/periph_rtc/ all term
make: Entering directory '/home/francisco/workspace/RIOT/tests/periph_rtc'
=== [ATTENTION] Testing Kconfig dependency modelling ===
Traceback (most recent call last):
  File "/home/francisco/workspace/RIOT/dist/tools/kconfiglib/genconfig.py", line 440, in <module>
    main()
  File "/home/francisco/workspace/RIOT/dist/tools/kconfiglib/genconfig.py", line 387, in main
    kconf = kconfiglib.Kconfig(args.kconfig_filename, warn_to_stderr=False)
  File "/home/francisco/workspace/RIOT/dist/tools/kconfiglib/kconfiglib.py", line 947, in __init__
    self._init(filename, warn, warn_to_stderr, encoding)
  File "/home/francisco/workspace/RIOT/dist/tools/kconfiglib/kconfiglib.py", line 1085, in _init
    self._parse_block(None, self.top_node, self.top_node).next = None
  File "/home/francisco/workspace/RIOT/dist/tools/kconfiglib/kconfiglib.py", line 2989, in _parse_block
    prev = self._parse_block(None, parent, prev)
  File "/home/francisco/workspace/RIOT/dist/tools/kconfiglib/kconfiglib.py", line 2922, in _parse_block
    self._parse_error("missing or bad symbol name")
  File "/home/francisco/workspace/RIOT/dist/tools/kconfiglib/kconfiglib.py", line 3879, in _parse_error
    raise KconfigError("{}error: couldn't parse '{}': {}".format(
kconfiglib.KconfigError: bin/native/generated/Kconfig.dep:1: error: couldn't parse 'config': missing or bad symbol name
make: *** No rule to make target '/home/francisco/workspace/RIOT/tests/periph_rtc/bin/native/generated/out.config', needed by 'check-kconfig-errors'.  Stop.
make: Leaving directory '/home/francisco/workspace/RIOT/tests/periph_rtc'
```

**Comment by fjmolinas** [comment]
> Bumped into the following when compiling with TEST_KCONFIG 1, then 0, then 1:

`TEST_KCONFIG =1 make clean` followed by `TEST_KCONFIG =1 make all` fixed the issue, but not `TEST_KCONFIG =1 make clean all`, seems like there is something wrong with the targets ordering.

**Comment by fjmolinas** [comment]
> Check that the dependencies are correctly modelled

I had a close look, everything seems ok.

> Peripheral test applications should still work as usual when using TEST_KCONFIG=1

I compared the binaries on some applications they are the same

> Non-supported applications should explain the missing features/modules.

```
TEST_KCONFIG=1 BOARD=native make -C tests/periph_rtt/ clean all
make: Entering directory '/home/francisco/workspace/RIOT/tests/periph_rtt'
=== [ATTENTION] Testing Kconfig dependency modelling ===
[genconfig.py]:ERROR-=> The module MODULE_PERIPH_RTT could not be set to y.
[genconfig.py]:ERROR-   Check the following unmet dependencies: HAS_PERIPH_RTT (=n)

...
...

TEST_KCONFIG=1 BOARD=native make -C tests/periph_backup_ram
make: Entering directory '/home/francisco/workspace/RIOT/tests/periph_backup_ram'
=== [ATTENTION] Testing Kconfig dependency modelling ===
[genconfig.py]:ERROR-=> The application symbol (APPLICATION) is not set.
[genconfig.py]:ERROR-   Check that the symbol defaults to 'y'.
[genconfig.py]:ERROR-   Check the following unmet dependencies: HAS_BACKUP_RAM (=n)
```

> Usage of Kconfig as usual (without TEST_KCONFIG=1) should still work normally

The only issue I found seems unrelated to this PR https://github.com/RIOT-OS/RIOT/pull/15421#issuecomment-728837002.

**Comment by fjmolinas** [comment]
@leandrolanzieri please squash once https://github.com/RIOT-OS/RIOT/pull/15421#discussion_r525039177 is addressed, https://github.com/RIOT-OS/RIOT/pull/15421#issuecomment-728837002 can be addressed in another pr.

File: drivers/periph_common/Kconfig (hunk 1 above)
Comment (leandrolanzieri) [reply drivers/periph_common/Kconfig]: Moved the RTC-related symbols to its own file

**Review by fjmolinas — APPROVED** [review]
ACK!

**Comment by leandrolanzieri** [comment]
Rescheduled CI to give priority to #15455

**Pull request merged**